draft-documents/encrypted-data-vaults.md

```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
-interface; pointers to external services.
-
-At a lower level, [Tahoe-LAFS](https://tahoe-lafs.org/trac/tahoe-lafs) uses a
-client-server architecture, whereby a local client encrypts data, breaks it into
-pieces, and stores it on one or more servers on other devices. The client
-creates and stores redundant pieces of data so a system is resilient to
-corruption of some shards or servers going offline. Servers can neither read nor
-modify the data. Data is organized in a filesystem like directory structure and
-access control makes use of this. Data can be mutable or immutable, and is
-always encrypted.
-
-Datashards is concerned with encrypted storage and replication of data across
-devices. Datashards is similar to Tahoe-LAFS, but uses a more generalized URI
-scheme and object capabilities for sharing of data objects. Data is always
-encrypted.
+users to sign into client applications, which can then read or write data.
+Resources (data objects) on Solid servers are represented by HTTP URIs, and Solid
+servers receive HTTP requests containing RDF payloads and create or modify the
+target URI accordingly.
+
+Data Hubs uses JSON Web Tokens (JWTs) and specified endpoints. Multiple requests
+are required, first to retrieve metadata for the desired data object(s), and then
+to retrieve the sequence of commits which make up the actual data. Mechanisms for
+authentication are still under development. Access control is carried out via
+posting to the Permissions interface.
+
+Tahoe-LAFS uses a client-server architecture, whereby a local client encrypts
+data, breaks it into pieces, and stores it on one or more servers on other devices.
+Services are identified with [Foolscap](https://foolscap.lothar.com/trac) URIs
+and the client can be configured to use HTTP, (S)FTP, or listen to a local
+directory ('magic folder') to create, update and delete data. Data is organized
+in a filesystem like directory structure and access control makes use of this.
 
 IPFS is a distributed content-addressed storage mechanism which breaks data up
 into Merkel-DAGs. IPFS uses [IPLD](https://github.com/ipld/specs) to generate
 URIs for data from the content, and to link content together on the network; and
-uses DHTs to discover content on the network. IPFS does not encrypt data, and
-stores it across a public network.
-
-[Hyperledger Aries](https://github.com/hyperledger/aries-rfcs) is an
-infrastructure for agent interactions but does not provide a solution for data
-storage.
+uses DHTs to discover content on the network.
+
+### Indexing and Querying
+
+Encrypted data which is opaque to the storage server introduces challenges for
+indexing and searching the contents of the data. Some systems work around this
+with a certain amount of unencrypted metadata attached to the data objects.
+Another possibility is unencrypted listings of pointers to filtered subsets
+of data.
+
+Resources in Solid are listed by URI in `ldp:Container`s, which serve as indexes
```

Comment:
I think referencing `ldp:Container` might be too detailed for the overview. Maybe something like:

Like Nextcloud, Solid attempts to provide a web accessible interface to a file system. Resources (linked data or arbitrary files) are organized into folder-like Containers. Precisely how the data is stored is an implementation detail ... [etc]